### ISSUE TITLE
Use PixelCopy to create Screenshots

### ISSUE THREAD

**Issue by kahest** [getsentry/sentry-java#2916]
### Description

Using PixelCopy to create screenshots is likely to provide better performance than the current approach.
See https://developer.android.com/reference/android/view/PixelCopy

**Comment by markushi** [comment]
So I've compared the old API against the new one and it's slower.

Turns out the async API is a lie, only the callback gets executed async - but capturing is happening on the calling thread.
See [the source for more details.](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/graphics/java/android/view/PixelCopy.java;l=186;drc=9bdd2e6151aa7a70bb4a12e91e7f68959b6334cb) 

Surprisingly `PixelCopy` is also around 20% slower, maybe because it e.g. also draws shadows?
Pixel 6a, Demo App, Release build
> n=20, PixelCopy.request avg 145.6ms, ScreenshotUtils.takeScreenshot avg 116.85ms
